src/neuralnet/neuron_layer/dummy.cc

```diff
@@ -71,4 +78,49 @@ void DummyLayer::ComputeGradient(int flag, const vector<Layer*>& srclayers) {
     Copy(grad_, srclayers[0]->mutable_grad(this));
 }
 
+void DummyLayer::Feed(vector<int> shape, vector<float>* data, int op){
```

Comment:
Would it be better to change the signature of Feed to

    Feed(const vector<int>& shape, const vector<float*>& data, vector<int>& aux_data);

Currently, every layer has a `data_` and `aux_data_` fields, including the dummy layer.